test/browser/events.test.jsx

```diff
@@ -291,4 +291,29 @@ describe('event handling', () => {
 			false
 		);
 	});
+
+	it('should support scroll snap events', () => {
+		let scrollSnapChange = vi.fn();
+		let scrollSnapChanging = vi.fn();
+
+		render(
+			<div
+				onScrollSnapChange={scrollSnapChange}
+				onScrollSnapChanging={scrollSnapChanging}
+			/>,
+			scratch
+		);
+
+		expect(proto.addEventListener).toHaveBeenCalledTimes(2);
+		expect(proto.addEventListener).toHaveBeenCalledWith(
+			'scrollsnapchange',
+			expect.any(Function),
+			false
+		);
+		expect(proto.addEventListener).toHaveBeenCalledWith(
+			'scrollsnapchanging',
+			expect.any(Function),
+			false
+		);
+	});
```

Comment:
As this is a types-only change, we don't really need a new test case here. There's nothing particularly interesting about these two as far as I can see (as opposed to `focusin`/`focusout`, which have been a bit iffy, and `pointercapture` which caused a hiccup for our regex)